docs/validators/staking-v4.md

```diff
@@ -0,0 +1,338 @@
+
+The selection process for the best X nodes from the auction into the "to be shuffled into waiting" list can be executed
+in two ways.
+
+**The first** and easy version is a strict selection where all nodes from the auction list are sorted based on their
+topUp, and only the first ones are selected.
+
+This approach is strict, requiring validators to manage their number of nodes if their topUp is close to the selection
+threshold. The topUp is computed as TotalStake/NumberOfNodes - 2.5k eGLD. If a validator registers more nodes, their
+topUp per node decreases. For example:
+
+- 10 nodes, 50k eGLD = 2.5k topUp per node
+- 20 nodes, 50k eGLD = 0 topUp
+
+In this scenario, if a staking provider registers all 20 nodes, they will be at the end of the list in case of an
+auction selection. If only 320 nodes out of 340 can be selected, and every other node has at least 1 eGLD topUp on each
+of them, none of the 20 nodes from this staking provider will be selected.
+
+Since this selection occurs at the end of every epoch, staking providers near the topUp limit must continually monitor
+and adjust their nodes, unstaking or staking nodes based on the topUp per node of other providers. Sorting nodes based
+on topUp does not provide adequate protection for staking providers, requiring constant supervision and action.
+
+**The second version**, currently implemented and explained in the following chapters, addresses the shortcomings of the
+first version.
+
+# **Current Implementation**
+
+:::note
+Please note that the numbers below are indicative and only used to better exemplify the model.
+:::
+
+In the current implementation (staking 3.5), we have:
+
+1. A capped number of 3200 **nodes in the network**, including:
+    - 1600 active/eligible validators globally, split into 400 nodes per shard
+    - 1600 waiting validators globally, split into 400 nodes per shard
+2. An uncapped `FIFO` queue where newly staked nodes are placed and await participation in the network.
+
+:::important
+
+Currently, a queued node can participate in the network only if an existing node is either unstaked or jailed.
+
+:::
+
+![Current Staking](/validators/stakingV4/current-staking.png)
+
+Nodes are distributed in the following steps:
+
+1. Randomly shuffle out 80 eligible validators for each shard, resulting in 320 (80 validators per 4 shards)
+   shuffled-out validators.
+2. Select these 320 shuffled-out validators to be randomly but evenly distributed at the end of each shard's waiting
+   list.
+3. For each shard, replace the previously shuffled-out validators with 80 waiting validators from the top of each
+   shard's waiting list.
+
+In the current implementation, each node, regardless of its top-up, has equal chances of participating in the consensus.
+However, starting with staking phase 4, validators with higher staked top-up will have higher chances of participation.
```

Comment:
actually those nodes with no topup, or small topup will have really small chance to enter into validation.

Reply:
Changed text to:

"Starting with staking phase 4, the probability of validators entering the validation process will be significantly influenced by the amount of their staked top-up. Validators with a higher staked top-up will have considerably greater chances of participation, while those with little or no top-up will find their chances of entering into validation markedly reduced."